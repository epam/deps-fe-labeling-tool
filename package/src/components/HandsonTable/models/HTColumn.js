
import PropTypes from 'prop-types'

class HTColumn {
  constructor (data) {
    this.data = data
  }
}

const htColumnShape = PropTypes.shape({
  data: PropTypes.string.isRequired
})

export {
  HTColumn,
  htColumnShape
}
