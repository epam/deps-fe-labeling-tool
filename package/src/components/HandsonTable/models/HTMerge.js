import PropTypes from 'prop-types'

class HTMerge {
  constructor (row, col, rowspan, colspan) {
    this.row = row
    this.col = col
    this.rowspan = rowspan
    this.colspan = colspan
  }
}

const htMergeShape = PropTypes.shape({
  row: PropTypes.number.isRequired,
  col: PropTypes.number.isRequired,
  rowspan: PropTypes.number.isRequired,
  colspan: PropTypes.number.isRequired
})

export {
  HTMerge,
  htMergeShape
}
