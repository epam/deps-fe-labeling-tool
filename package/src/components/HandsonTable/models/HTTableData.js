
import PropTypes from 'prop-types'

const rowOfPrimitivesShape = PropTypes.arrayOf(
  PropTypes.oneOfType([
    PropTypes.number.isRequired,
    PropTypes.string.isRequired
  ])
)

const rowsOfPrimitivesShape = PropTypes.arrayOf(
  rowOfPrimitivesShape
)

export { rowsOfPrimitivesShape }
