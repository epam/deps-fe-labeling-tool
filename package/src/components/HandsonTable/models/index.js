
export * from './HTTableData'
export * from './HTMerge'
export * from './HTColumn'
