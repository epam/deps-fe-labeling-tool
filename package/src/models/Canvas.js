import PropTypes from 'prop-types'

export const canvasShape = PropTypes.shape({
  add: PropTypes.func,
  backgroundColor: PropTypes.string,
  defaultCursor: PropTypes.string,
  discardActiveObject: PropTypes.func,
  findTarget: PropTypes.func,
  fire: PropTypes.func,
  fxRemove: PropTypes.func,
  getActiveObject: PropTypes.func,
  getActiveObjects: PropTypes.func,
  getCenter: PropTypes.func,
  getHeight: PropTypes.func,
  getObjects: PropTypes.func,
  getPointer: PropTypes.func,
  getWidth: PropTypes.func,
  getZoom: PropTypes.func,
  height: PropTypes.number,
  off: PropTypes.func,
  on: PropTypes.func,
  remove: PropTypes.func,
  renderAll: PropTypes.func,
  requestRenderAll: PropTypes.func,
  selection: PropTypes.bool,
  sendToBack: PropTypes.func,
  setActiveObject: PropTypes.func,
  setBackgroundColor: PropTypes.func,
  setBackgroundImage: PropTypes.func,
  setCursor: PropTypes.func,
  setHeight: PropTypes.func,
  setViewportTransform: PropTypes.func,
  setWidth: PropTypes.func,
  setZoom: PropTypes.func,
  viewportTransform: PropTypes.arrayOf(PropTypes.number),
  width: PropTypes.number,
  zoom: PropTypes.number,
  zoomToPoint: PropTypes.func
})
