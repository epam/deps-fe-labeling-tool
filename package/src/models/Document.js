import PropTypes from 'prop-types'
import { KnownLanguage } from '@/enums/KnownLanguage'

class Document {
  constructor (pages, language = null, name = '', engine = '', extraName = '') {
    this.pages = pages
    this.language = language
    this.name = name
    this.engine = engine
    this.extraName = extraName
  }
}

const documentShape = PropTypes.shape({
  language: PropTypes.oneOf(
    Object.values(KnownLanguage)
  ),
  pages: PropTypes.arrayOf(PropTypes.string).isRequired,
  name: PropTypes.string.isRequired,
  engine: PropTypes.string.isRequired,
  extraName: PropTypes.string
})

export {
  documentShape,
  Document
}
