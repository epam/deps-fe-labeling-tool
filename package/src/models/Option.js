import PropTypes from 'prop-types'

class Option {
  constructor (value, text, disabled) {
    this.value = value
    this.text = text || value
    this.disabled = disabled
  }
}

const optionShape = PropTypes.shape({
  text: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  disabled: PropTypes.bool
})

export {
  Option,
  optionShape
}
