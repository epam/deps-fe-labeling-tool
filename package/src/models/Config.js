import PropTypes from 'prop-types'
import { documentShape } from '@/models/Document'
import { fieldShape } from '@/models/Field'
import { markupShape } from '@/models/Markup'
import { ocrEngineShape } from '@/models/OcrEngine'
import { ocrLanguageShape } from '@/models/OcrLanguage'
import { settingsShape } from '@/models/Settings'

class Config {
  constructor (document, fields, ocr, api, markup, settings, events) {
    this.document = document
    this.fields = fields
    this.ocr = ocr
    this.api = api
    this.markup = markup
    this.settings = settings
    this.events = events
  }
}

const ocrShape = PropTypes.shape({
  engines: PropTypes.arrayOf(ocrEngineShape),
  languages: PropTypes.arrayOf(ocrLanguageShape)
})

const apiShape = PropTypes.shape({
  close: PropTypes.func.isRequired,
  getImage: PropTypes.func,
  save: PropTypes.func.isRequired,
  saveMarkup: PropTypes.func,
  ocrTable: PropTypes.func,
  ocrText: PropTypes.func,
  addFieldForm: PropTypes.func
})

const eventsShape = PropTypes.shape({
  onClose: PropTypes.func
})

const configShape = PropTypes.shape({
  document: documentShape.isRequired,
  fields: PropTypes.arrayOf(fieldShape).isRequired,
  ocr: ocrShape,
  api: apiShape.isRequired,
  markup: markupShape,
  settings: settingsShape,
  events: eventsShape
})

export {
  apiShape,
  configShape,
  Config
}
