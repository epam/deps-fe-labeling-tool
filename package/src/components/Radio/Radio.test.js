import React from 'react'
import { shallow } from 'enzyme'
import { Option } from '@/models/Option'
import { Radio, RadioGroup } from './'

describe('Component: RadioGroup', () => {
  let wrapper, defaultProps

  beforeEach(() => {
    defaultProps = {
      onChange: jest.fn(),
      options: [new Option('hello', 'world', false)]
    }
    wrapper = shallow(<RadioGroup {...defaultProps} />)
  })

  it('should render correct layout based on props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call props.onChange in case calling to RadioGroup onChange', () => {
    const mockEvent = {
      target: {
        value: 'hello world'
      }
    }
    const radioGroupProps = wrapper.props()
    radioGroupProps.onChange(mockEvent)
    expect(defaultProps.onChange).nthCalledWith(1, mockEvent.target.value)
  })
})

describe('Component: Radio', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Radio />)
  })

  it('should render correct layout based on props', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
