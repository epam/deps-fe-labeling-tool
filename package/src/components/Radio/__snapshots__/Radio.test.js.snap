// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Radio should render correct layout based on props 1`] = `<Radio />`;

exports[`Component: RadioGroup should render correct layout based on props 1`] = `
<Memo(ForwardRef)
  onChange={[Function]}
>
  <Radio
    key="0"
    value="hello"
  >
    world
  </Radio>
</Memo(ForwardRef)>
`;
